src/CSLOLUtils.cpp

```diff
@@ -118,10 +118,84 @@ bool CSLOLUtils::isUnnecessaryAdmin() {
     }
     return result;
 }
-#else
-// TODO: macos implementation
-QString CSLOLUtils::detectGamePath() { return ""; }
+#elif __APPLE__
+#    include <libproc.h>
+#    include <string.h>
+
+#    include <filesystem>
+#    include <iostream>
+
+QString CSLOLUtils::detectGamePath() {
+    pid_t *pid_list;
+
+    //
+    //  Get necessary size for `pid_list`
+    //
+    int bytes = proc_listallpids(nullptr, 0);
+
+    //
+    //  Something went wrong trying
+    //  to find all the processes.
+    //
+    if (bytes <= 0) {
+        return "";
+    }
+
+    pid_list = (pid_t *)malloc(bytes * sizeof(pid_t));
+
+    bytes = proc_listallpids(pid_list, bytes * sizeof(pid_t));
+
+    for (int i = 0; i < bytes / sizeof(pid_list[0]); i++) {
+        struct proc_bsdinfo proc;
+
+        //
+        //  Info about this PID.
+        //
+        int st = proc_pidinfo(pid_list[i], PROC_PIDTBSDINFO, 0, &proc, PROC_PIDTBSDINFO_SIZE);
+
+        //
+        //  We're looking for a LeagueClient PID.
+        //
+        if (strcmp(proc.pbi_name, "LeagueClient") == 0) {
+            char game_path[PROC_PIDPATHINFO_MAXSIZE];
+
+            //
+            //  Find the parent path
+            //  of the LeagueClient PID.
+            //
+            int len = proc_pidpath(proc.pbi_pid, game_path, sizeof(game_path));
+
+            //
+            //  Something went wrong.
+            //
+            if (len == 0)
```

Comment:
Can this also return negative value, please check?

Reply:
As far as I can tell: no.

```cpp
int
proc_pidpath(int pid, void * buffer, uint32_t  buffersize)
{
	int retval, len;

	if (buffersize < PROC_PIDPATHINFO_SIZE) {
		errno = ENOMEM;
		return(0);
	}
	if (buffersize >  PROC_PIDPATHINFO_MAXSIZE) {
		errno = EOVERFLOW;
		return(0);
	}

	retval = __proc_info(PROC_INFO_CALL_PIDINFO, pid, PROC_PIDPATHINFO,  (uint64_t)0,  buffer, buffersize);
	if (retval != -1) {
		len = (int)strlen(buffer);
		return(len);
	}
	return (0);
}
```

Source: <https://opensource.apple.com/source/xnu/xnu-2782.40.9/libsyscall/wrappers/libproc/libproc.c>